### PR TITLE
Custom implementation for object hashing

### PR DESCRIPTION
**Pull request by tkupka**
Allows custom implementation for object hashing to detect changes in
unordered collection of value objects

https://github.com/javers/javers/issues/1301

### CODE REVIEW

File: javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        //when
+        Diff diff = javers.compareCollections(oldSet, newSet, Person.class);
+
+        System.out.println(diff.prettyPrint());
+
+        assertThat(diff.getChanges()).hasSize(5);
+
+        //then
+        //there should be one change of type {@link ValueChange}
```
Comment (bartoszwalacik) [inline javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java:39]: there should be one change ?

File: javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import org.javers.core.examples.vo.Person;
+import org.javers.core.graph.HashCodeObjectHasher;
+import org.junit.Test;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+/**
+ * @author tomas.kupka
+ */
+public class ComparingTopLevelSetVoExample {
```
Comment (bartoszwalacik) [inline javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java:20]: please rewrite this test in Groovy

File: javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java (hunk 1 above)
Comment (tkupka) [reply javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java:39]: Definitely not. It should be 4. Just remained from the original test down below

File: javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java (hunk 2 above)
Comment (tkupka) [reply javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java:24]: done

**Comment by bartoszwalacik** [comment]
thanks @tkupka , your PR is merged from this PR https://github.com/javers/javers/pull/1365

**Pull request closed without merging**